### DISCUSSION TITLE
External authentication for embeded videos

### DISCUSSION THREAD

**Discussion by zynex-mbrunner** [mediacms-io/mediacms#674] [Q&A]
I would like to list the available videos on a external site via the api only for authenticated users on the external site.

Is it possible to add a user token as a parameter to the embed endpoint?
Or is it possible to authenticate a user via api and then redirect to the embed endpoint?

**Comment by mgogoulos** [comment]
can you elaborate more on what you're trying to achieve? Give a detailed example and there might be suggestions!

**Comment by zynex-mbrunner** [comment]
Following scenario:
The video content is managed directly on the mediacms server. The server would be configured to allow only authenticated users and no public viewing of the content.

On different site the videos will be presented to a closed user group (authenticated with local users there). The list of the available videos will be pulled over the REST API. Now the authenticated user should be able to view the video without logging in again on the mediacms server. 

I see 2 ways to achieve this:

Possibility 1: Using the API create a token that allows the user to view the selected video. Generate the URL for the player with the token of the video and the access token.

Possibility 2: Sync all users via the api to the mediacms server. When a user requests to view a video the external site should be able to log the user in on the mediacms server and then redirect to the viewer.

One question: I can see that on the API is an endpoint /login. But where can I use later on the received token?

**Reply by mgogoulos** [reply]
I believe that could work with a custom authentication (as the example on https://www.django-rest-framework.org/api-guide/authentication/ ), a simple scenario could be that the authentication is through a GET variable and not as a HEADER variable in this example, thus the GET variable can be appended to all of the APIs for the media (example /api/v1/media/x?token=yz , so you are appending ?token=yz on all of the APIs). You then have to set MediaCMS to use this authentication along with the rest of it, and implement this authentication - it could be the case that you create manually a user, user_remote, and you make the custom authentication checkout the token on the GET parameters and if it's the correct one return that user. Thus all users on the remote site get to authenticate through the same user. 
Three downsides here would be that a) all users are supposed to authenticate as the same user, b) if a URL containing this token is exposed, others can view the media as well, c) on the remote site, you have to append the ?token=yz on the media. Depending on your use case, these might or not be issues. Last, there might be simpler solutions out of the box with some sort of integration (or even setting) for Django Rest Framework

**Comment by marcofloriano** [comment]
So @mbrunner77 how did you solved that ?
